tests/unit/utils/test_vt.py

```diff
@@ -16,19 +16,34 @@
 import random
 import subprocess
 import time
+import io
 
 # Import Salt Testing libs
 from tests.support.unit import TestCase, skipIf
+from tests.support.paths import CODE_DIR
 
 # Import Salt libs
+import salt.utils
 import salt.utils.files
 import salt.utils.platform
 import salt.utils.vt
+import salt.utils.stringutils
 
 # Import 3rd-party libs
 from salt.ext.six.moves import range  # pylint: disable=import-error,redefined-builtin
 
 
+def stdout_fileno_available():
```

Comment:
I think the cleanest option would be to put this test, the switch from the current stdout and stderr if necessary ( https://github.com/saltstack/salt/pull/53404/files#diff-64e62f3c1ef41b8920ec7488db35ec28R246-R252 ) and the restore ( https://github.com/saltstack/salt/pull/53404/files#diff-64e62f3c1ef41b8920ec7488db35ec28R246-R252 ) in a decorator. I'm not sure whether I should put the decorator code in this file since it's probably only needed here or whether to add it to one of the libraries of decorators that already exist.

The intent is to report given the environment the code is executing in whether you will get an exception if you try and call sys.stdout.fileno() at this point. Since we do modify the environment such that this will produce different results if called at different points it's safer to not cache the result since the result would be wrong if someone fails to properly restore the environment after changing it. It's not a particularly expensive operation and it isn't called that often.